src/http/server/handlers/static_file_handler.cr

```diff
@@ -185,7 +185,18 @@ class HTTP::StaticFileHandler
       else
         finish = finish_string.to_i64? || return
       end
-      unless start
+      if file_size.zero?
+        # > When a selected representation has zero length, the only satisfiable
+        # > form of range-spec in a GET request is a suffix-range with a non-zero suffix-length.
+
+        # This return value signals an unsatisfiable range.
+        if start
```

Comment:
This doesn't reject ranges like `bytes=-0` or even `bytes=--2`; should be `if start || finish <= 0`